### PR TITLE
Feature/diff matrix download

### PR DESCRIPTION
**Pull request by rubenchazarra**
This PR includes: 
- Updated data import step based on [cell-types-train-control-workflow] (https://github.com/ebi-gene-expression-group/cell-types-train-control-workflow). Enabling the following features: 
--> Data import params is specified in a txt file
--> Differential matrix type download 
- Manual data input for cross validation. 
--> Input are zipped directories with all the required files 
--> Logic is thought to enable any kind of matrix type to be passed

### CODE REVIEW

File: envs/load_data.yaml [hunk 1]
```diff
@@ -5,4 +5,4 @@ channels:
   - defaults
   - ebi-gene-expression-group
 dependencies:
-  - atlas-data-import=0.0.4
+  - atlas-data-import=0.0.5
```
Comment (a-solovyev12) [inline envs/load_data.yaml]: the latest version is 0.0.9 I believe, you could switch directly to it

File: nextflow.config [hunk 2]
```diff
@@ -20,145 +20,142 @@ profiles {
...
-
-        ref_acc_code = "E-MTAB-6386"
-        ref_output_dir = "ref_data"
-        ref_num_clust = 11
-
-        query_acc_code = "E-MTAB-6386"
-        query_output_dir = "query_data"
-        query_num_clust = 11 
-    }
+	label_analysis_outdir ="${baseDir}/data/tools_outputs"
```
Comment (a-solovyev12) [inline nextflow.config:25]: @rubenchazarra  is just the tab width that is changed here?

Comment (rubenchazarra) [reply nextflow.config:25]: Yep, only the new data_import slot is added. The rest remains the same @a-solovyev12

File: main.nf [hunk 3]
```diff
@@ -3,119 +3,153 @@
...
+	                    .filter{ it != null }
+	                    .unique()
+
+	// parse txt file with dataset accessions into a queue channel; build combinations with matrix types 
+	IMPORT_PARAMS = Channel
+	                .fromPath(params.data_import.training_datasets)
+	                .splitCsv(header:false, sep:",")
+	                .combine(UNIQUE_MATRIX_TYPES)
+
+process fetch_training_datasets {
```
Comment (a-solovyev12) [inline main.nf:29]: could you make sure the code is at a consistent level of indentation?

File: main.nf [hunk 4]
```diff
@@ -3,119 +3,153 @@
...
-    query_data = params.data_download.query_output_dir
-    query_n_clust = params.data_download.query_num_clust.toString()
-    query_markers = "marker_genes_" + query_n_clust + ".tsv"
+// download data
+if(params.data_import.run == "True"){
+
+	// initialise required channels 
+	DATASET_IDS = Channel.create()
+	NUM_CLUST = Channel.create()
+	BARCODE_COLUMN = Channel.create()
```
Comment (a-solovyev12) [inline main.nf]: Seems like you're not using the above channels (except `NUM_CLUST`) anywhere else in the script. Is that supposed to be so? If yes, I think we can get rid of them.

Comment (rubenchazarra) [reply main.nf]: Right, none of them are used later on (as NUM_CLUST is empty, it's N_CLUST the one carrying the value). I'm erasing them

Comment (rubenchazarra) [reply main.nf]: Actually, I copied the chunk from [here](https://github.com/ebi-gene-expression-group/cell-types-train-control-workflow/blob/bf0b9cb5b8ce8615768ba83a85deea73e9c1c7ff/main.nf#L4), but what are these channels for?

File: main.nf [hunk 5]
```diff
@@ -3,119 +3,153 @@
...
+	    SCPRED_DATA
+	}
+	
+	// add number of clusters to Garnett data 
+	GARNETT_FULL_DATA = GARNETT_DATA.merge(N_CLUST)
+
+}else{
+	// manual data input (Cross-Validation)
+	
+	MANUAL_INPUT_DATA = Channel.fromPath("${params.input_data}/*.zip")
```
Comment (a-solovyev12) [inline main.nf]: so you're assuming that data will be provided as an archive?

Comment (rubenchazarra) [reply main.nf]: This is the input coming from the CV workflow, which comes into zipfiles. To make it even more generic I could maybe say: 
```
else{
   if(cross_validation_input == "True"){do what I'm doing}
   if(manual_data_input == "True"){ different structure}
}
```

Comment (a-solovyev12) [reply main.nf]: @rubenchazarra If that's not adding too much complexity, go for it. Thanks!

File: main.nf [hunk 6]
```diff
@@ -3,119 +3,147 @@
...
+	# unzip test 
+	test_zipdir=\$(unzip -qql $test_zip | head -n1 | tr -s ' ' | cut -d' ' -f5- | sed 's|/.*||')
+	unzip $test_zip
+	# unzip train 
+	unzip $train_zip
+	"""
+	}
+	
+	// extract marker's number of clusters value	
+	MARKERS.map{it -> it.getBaseName().toString().split('\\_')[2] }.first().set{N_CLUST}
```
Comment (a-solovyev12) [inline main.nf:128]: Why not use information about the number of clusters that you extract from the initial file with datasets? Actually, the chances are we won't need them after the marker files are updated.

Comment (rubenchazarra) [reply main.nf:128]: When passing cross-validation data from the parent workflow onto the child, CSV file does not get read, and hence cluster number has to be extracted from the parent input.zip files

**Comment by a-solovyev12** [comment]
@rubenchazarra feel free to merge after addressing the feedback. Thanks!

**Review by a-solovyev12 — APPROVED** [review]

**Pull request merged**